src/gasPriceOracle/adapters/op.ts

```diff
@@ -0,0 +1,17 @@
+import assert from "assert";
+import { BigNumber, providers } from "ethers";
+import { chainIsOPStack } from "../../utils";
+import { GasPriceEstimate } from "../types";
+
+// 50 Wei should be sufficient to land a transaction on the OP stack.
+// Double it just to be on the safe side. The user is encouraged to scale it further as needed.
+const MAX_PRIORITY_FEE_PER_GAS = BigNumber.from("100");
+
+export async function legacy(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
+  assert(chainIsOPStack(chainId), `Unsupported OP chain ID: ${chainId}`);
+  const gasPrice = await provider.getGasPrice(); // Get the base L2 fee.
```

Comment:
Does this gas price not include the priority fee already?

Reply:
Yep, you're right. I was relying on some documentation from [Chainstack](https://docs.chainstack.com/reference/ethereum-getgasprice) that says eth_gasPrice returns the base fee, but when I tested it, I see the following:

`eth_gasPrice = baseFeePerGas + eth_maxPriorityFeePerGas`

I have a script that reproduces this if you're interested to try it:

```
#!/bin/sh

# Define an RPC URL here.
NODE_URL='...'

ETH_GETBLOCK='{ "id": 1, "jsonrpc": "2.0", "method": "eth_getBlockByNumber", "params": ["latest", false] }'
ETH_GASPRICE='{ "id": 1, "jsonrpc": "2.0", "method": "eth_gasPrice" }'
ETH_MAXPRIORITYFEEPERGAS='{ "id": 1, "jsonrpc": "2.0", "method": "eth_maxPriorityFeePerGas" }'

for query in "${ETH_GETBLOCK}" "${ETH_GASPRICE}" "${ETH_MAXPRIORITYFEEPERGAS}"; do
  result=`curl --request POST \
       --url "${NODE_URL}" \
       --header 'accept: application/json' \
       --header 'content-type: application/json' \
       --data "${query}" 2>/dev/null`
  echo ${result} | python3 -m json.tool
  echo "\n"
done
```

Then run it:
`sh rpc.sh  | grep -i -e result -e baseFeePerGas`

Reply:
So just to be clear, we get the maxFeePerGas for this transaction by querying `eth_gasPrice`, which itself is `baseFee + someDust`, and then we add an additional 100 wei? Do we just want the base fee or do we want this extra dust to be included?

Reply:
Ok so are we double counting the priority fee then?

Reply:
I've dug into it a bit more; in general I think we want to be calling `eth_maxPriorityFeePerGas` in parallel with `eth_gasPrice`. This ensures we can isolate the priority fee from the base fee.

ethers v5 doesn't permit us to call `eth_maxPriorityFeePerGas`. ethers v6 does, and viem also does.

I've found I can modify the existing viem adapter to do what I want, so I think I'll pivot this PR in that direction and run some testing to compare with the current gas estimation that we do.

Reply:
Worst case can we make a direct call to the rpc? Unless viem lets us do this natively

Reply:
Yep; ethers permits this via `provider.send()` (contingent upon provider being of type `JsonRpcProvider`. Viem also seems to permit it via `client.request()`, but there's some type coaxing that's needed to make it work.